.circleci/config.yml

```diff
@@ -177,6 +177,15 @@ jobs:
           name: debug | brew version
           command: |
             brew -v
+      - run:
+          name: brew update if needed # temporary solution for this https://discuss.circleci.com/t/macos-image-users-homebrew-brownout-2021-04-26/39872/2
+          command: |
+            # if the version is not 3.x.x, try brew update
+            ruby -e 'exit(1) if `brew -v` =~ /2\.\d+\.\d+/' ||\
+            brew update || \
+            # if brew update fails due to shallow clone, try unshallowing and then brew update again
+            git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core fetch --unshallow && \
+            brew update
```

Comment:
https://discuss.circleci.com/t/macos-image-users-homebrew-brownout-2021-04-26/39872/2

Apparently, CircleCI provides both old and new macOS images so that some builds succeed with Homebrew v3 others fails with Homebrew v2😞 This way `brew update` will be called only when macOS image that has Homebrew v2. This can fall back to unshallow command when initial brew update fails.

Reply:
This is insane 😛 Love it ❤️